packages/typescript/src/core/entities/span_status.ts

```diff
@@ -0,0 +1,117 @@
+import { SpanStatus as OTelStatus, SpanStatusCode as OTelSpanStatusCode } from '@opentelemetry/api';
+
+/**
+ * MLflow Span Status module
+ *
+ * This module provides the MLflow SpanStatusCode enum and SpanStatus class,
+ * matching the Python MLflow implementation.
+ */
+
+/**
+ * Enum for status code of a span
+ * Uses the same set of status codes as OpenTelemetry
+ */
+export enum SpanStatusCode {
+  /** Status is unset/unspecified */
+  UNSET = 'UNSET',
+  /** The operation completed successfully */
+  OK = 'OK',
+  /** The operation encountered an error */
+  ERROR = 'ERROR'
+}
+
+/**
+ * Status of the span or the trace.
+ */
+export class SpanStatus {
+  /**
+   * The status code of the span or the trace.
+   */
+  readonly statusCode: SpanStatusCode;
+
+  /**
+   * Description of the status. This should be only set when the status is ERROR,
+   * otherwise it will be ignored.
+   */
+  readonly description: string;
+
+  /**
+   * Create a new SpanStatus instance
+   * @param statusCode The status code - must be one of SpanStatusCode enum values
+   * @param description Optional description, typically used for ERROR status
+   */
+  constructor(statusCode: SpanStatusCode | string, description: string = '') {
```

Comment:
if we remove the `string` type here i think typescript should warn people when they try to set it as a plain string, then we don't need to do the validation. that being said, i guess in practice users can ignore the TS warning and set random strings lol. in such cases though, maybe we should rely on backend throwing an error?

```suggestion
  constructor(statusCode: SpanStatusCode, description: string = '') {
```